bvm/ballerina-core/src/main/java/org/ballerinalang/channels/DatabaseUtils.java

```diff
@@ -281,7 +281,7 @@ private static BValue getValue(ResultSet resultSet, BType bType) throws SQLExcep
             case TypeTags.STRING_TAG:
                 return new BString(resultSet.getString(2));
             case TypeTags.BYTE_TAG:
-                return new BByte(resultSet.getByte(2));
+                return new BByte(Byte.toUnsignedLong(resultSet.getByte(2)));
```

Comment:
Seems like we have to use Byte.toUnsignedLong every time with BByte constructor. Can we simplify this?

Reply:
Ack; Simplified via commit f3c8993 by adding a new BByte constructor that accepts a byte value.